src/main/java/edu/gatech/cs2340/trydent/Mouse.java

```diff
@@ -0,0 +1,144 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+
+/**
+ * A static class for ease of using mouse events.
+ *
+ */
+public class Mouse {
+    private static double x;
+    private static double y;
+
+    private static Set<MouseButton> down = new HashSet<>();
+    private static Set<MouseButton> downBuffer = new HashSet<>();
+    private static Set<MouseButton> pressed = new HashSet<>();
+    private static Set<MouseButton> pressedBuffer = new HashSet<>();
+    private static Set<MouseButton> released = new HashSet<>();
+    private static Set<MouseButton> releasedBuffer = new HashSet<>();
+
+    private static final Object POSITION_LOCK = new Object();
+    private static final Object REPLACE_LOCK = new Object();
+    private static final Object UPDATE_LOCK = new Object();
+
+    /**
+     * Gets the absolute position of the mouse with left as zero.
+     * @return the x position
+     */
+    public static double getMouseX() {
+        synchronized(POSITION_LOCK) {
```

Comment:
The lock is not for concurrentmodificationexception.
I was thinking of the case where only one of the coordinates got updated, which results in slight inconsistency.
The real problem occurs with word tearing though, which means a write to a double could cause impossible values.
Refer (1.7 but still applies): http://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html